### PR TITLE
Solves night profile issue

### LINKED ISSUE #11: Not able to create profile for overnight times
**Describe the bug**
A user can't create a profile for the time night of one day to morning of the next day. For eg 11:00 PM - 7:00AM. On choosing these times the app will show enter a valid time.

**To Reproduce**
1. Open quiet hours.
2. Try to create a new profile for the above time conditions.

**Expected behavior**
The user should be able to create a profile for overnight times and the app should work as expected.

**Would you like to work on the issue?**

- [ ] Yes
- [ ] No
- Other: Open for all

### PR DESCRIPTION
**Pull request by mouryac9**
Fixes #11 

Changes: Check conditions in NewProfileFragment.kt file were slightly changed to solve the issue.

Screenshots for the change:

### CODE REVIEW

File: app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt [hunk 1]
```diff
@@ -142,7 +142,11 @@ class NewProfileFragment : Fragment() {
                 while (i < 7) {
                     if (days[i]) {
                         sAlarm(i + 1, profile)
-                        eAlarm(i + 1, profile)
+                        if ((shr > ehr) || ((shr == ehr) && (smin > emin))) {
+                            eAlarm(i + 2, profile)
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: when i=6 and i+2=8, then do you think it will work?
Have a look at it.

Comment (mouryac9) [reply app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: I am so sorry, that condition completely slipped off my mind. I'll just put in another if-else, cause i can't change value of 'i' and other option is to declare another variable which will just increase app's size.

**Review by rob729 — CHANGES_REQUESTED** [review]
@mouryac9 Squash your commits into a single commit.

**Comment by rob729** [comment]
@mouryac9 Have you once tested the app? Is it working as expected?
I forgot to test it today so will try to test it tomorrow and if it works fine then I will merge the PR.

**Comment by mouryac9** [comment]
It worked fine for me. It'll be better if you test it in real time because it won't work if you try to change device's time after setting up profile.

File: app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt [hunk 2]
```diff
@@ -142,7 +142,15 @@ class NewProfileFragment : Fragment() {
                 while (i < 7) {
                     if (days[i]) {
                         sAlarm(i + 1, profile)
-                        eAlarm(i + 1, profile)
+                        if ((shr > ehr) || ((shr == ehr) && (smin > emin))) {
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: @mouryac9 I want to ask one thing. Why include the  ((shr == ehr) && (smin > emin) in the if condition?
The rest looks good to me and I tested it today and it worked as expected.

Comment (mouryac9) [reply app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: Initially it was there to check that starting time is not greater than ending time. But now I guess it won't be of much use now that there's a 12 hour restriction. I'll remove it.

File: app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt [hunk 3]
```diff
@@ -142,7 +142,15 @@ class NewProfileFragment : Fragment() {
                 while (i < 7) {
                     if (days[i]) {
                         sAlarm(i + 1, profile)
-                        eAlarm(i + 1, profile)
+                        if ((shr > ehr)) {
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: why double (( in the if condition?
Remove one of the () pair.

**Review by rob729 — APPROVED** [review]
@mouryac9 Thanks for contributing to Quiet Hours.

**Pull request merged**